### DISCUSSION TITLE
[bun init] Typescript -> Put bun-types Inside @types Folder

### DISCUSSION THREAD

**Discussion by mahdi-farnia** [oven-sh/bun#1007] [General]
As you know when a `@types/*` packages installed it goes to `node_modules/@types/*` folder. And every things works properly without `reference directive` (`/// <reference types="bun-types"`) syntax or adding `types` to tsconfig `compilerOptions`.

When i tried to init a new project using `bun init` every things worked properly except typings: `Cannot find name Bun`. Until I figured out that typescript did not find `bun-types/types.d.ts` file, and i add it to `compilerOptions` and error gone.

I think bun-types should goes to `@types` folder instead of node_modules top level.

**OS:** MacOS 12.5
**Version**: 0.1.7 (latest right now)
**Typescript Version**: 4.7.4

**Comment by xhyrom** [comment]
There are 2 ways:
- Add bun-types to DefinitelyTyped, which would make compatibility and maintanance much worse because we would have no control over it
- bun add would automatically move the package to @types

**Reply by mahdi-farnia** [reply]
second one is the best option ( i think )

**Reply by leesei** [reply]
`@types/bun` available, this should be closed
